### PR TITLE
Unique Token Dissociate

### PR DESCRIPTION
**Pull request by joan41868**
**Description:**

This PR adds additional check for dissociating unique tokens - 0 owned nfts, as well as refactoring the already implemented checks for **deletion**, **expiry**, **treasury dissociation** and **frozen rel** .

Adds E2E tests:
* `tokenDissociateHappyPath`
* `tokenDissociateFailsIfAccountOwnsUniqueTokens`

**Related issue(s):**

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by Daniel-K-Ivanov** [comment]
Let's create a PR against the Protobufs repo and add a comment for the Token Disassociate operation that explains the details around NFT token disassociate and the difference that Expired NON_FUNGIBLE_UNIQUE Tokens cannot be disassociated if the account has balance (change compared to FUNGIBLE disassociate)

File: hedera-node/src/main/java/com/hedera/services/records/TransactionRecordService.java [hunk 1]
```diff
@@ -70,39 +72,40 @@ public void includeChangesToToken(Token token) {
...
+	 * 		List of models of the changed relationship
 	 */
-	public void includeChangesToTokenRel(TokenRelationship tokenRel) {
-		if (tokenRel.getBalanceChange() == 0L || !tokenRel.hasCommonRepresentation()) {
-			return;
-		}
+	public void includeChangesToTokenRels(final List<TokenRelationship> tokenRels) {
+		Map<Id, TokenTransferList.Builder> transferListMap = new HashMap<>();
+		for (final var tokenRel : tokenRels) {
+			if (tokenRel.getBalanceChange() == 0L && !tokenRel.hasCommonRepresentation()) {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/records/TransactionRecordService.java:88]: Should be `||` instead of `&&`

**Review by Daniel-K-Ivanov — CHANGES_REQUESTED** [review]

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Review by qnswirlds — COMMENTED** [review]
Merged by #1777

**Pull request closed without merging**